chia/cmds/param_types.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal, InvalidOperation
+from typing import Any, Optional, Union
+
+import click
+
+from chia.cmds.units import units
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.bech32m import bech32_decode, decode_puzzle_hash
+from chia.util.config import selected_network_address_prefix
+from chia.util.ints import uint64
+from chia.wallet.util.address_type import AddressType
+
+
+class TransactionFeeParamType(click.ParamType):
+    """
+    A Click parameter type for transaction fees, which can be specified in XCH or mojos.
+    """
+
+    name: str = "uint64"  # output type
+    value_limit: Decimal = Decimal(0.5)
+
+    def convert(self, value: Any, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> uint64:
+        if isinstance(value, uint64):  # required by click
+            return value
+        mojos = False  # TODO: Add unit logic
+        if mojos:
+            try:
+                return uint64(int(value))
+            except ValueError:
+                self.fail("Fee must be positive integer value in mojos", param, ctx)
+        try:
+            d_value = Decimal(value)
+            if 0 > d_value:
+                self.fail("Fee can not be negative", param, ctx)
+            if self.value_limit != Decimal(0) and d_value > self.value_limit:
+                self.fail(f"Fee must be in the range 0 to {self.value_limit}", param, ctx)
+            return uint64(int(d_value * units["chia"]))
```

Comment:
Some reason this doesn't work?  We used to have hinting issues around this but they are (mostly) fixed.

```suggestion
            return uint64(d_value * units["chia"])
```